streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -874,6 +876,53 @@ void to(final String topic,
     void to(final TopicNameExtractor<K, V> topicExtractor,
             final Produced<K, V> produced);
 
+    /**
+     * Convert this stream to a {@link KTable}.
+     * <p>
+     * Note that this is a logical operation and only changes the "interpretation" of the stream, i.e., each record of
+     * it was a "fact/event" and is re-interpreted as update now (cf. {@link KStream} vs {@code KTable}).
+     *
+     * @return a {@link KTable} that contains the same records as this {@code KStream}
+     */
+    KTable<K, V> toTable();
```

Comment:
Given the change that we might insert a repartition topic in the key was change upstream, we should extend the JavaDocs accordingly. Compare the JavaDocs of `groupByKey()` that contain a paragraph about auto-repartitioning. We should add a similar (or maybe even the exact some if suitable) for all four overloads of `toTable()`, too.